### PR TITLE
Migration to a global state and redux

### PR DESCRIPTION
**Pull request by alan-maldonado**
**What does this PR do?**
In order to make this application more scalable and easy to introduce new features this PR provides a global state to handle with multiple reducers the information of the application.
- Global state
- Add middleware to analyze global state
screen shot 2017-05-21 at 9 47 24 pm
- Merge list questions component and answered list component
- Added alert system
screen shot 2017-05-21 at 9 49 13 pm
- Added some TODO of nice to have technical functionality.

Sorry for long PR, but it is a major change.

### CODE REVIEW

File: src/components/ListQuestions.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        </Modal.Body>
+        <Modal.Footer>
+          <button className="btn btn-dark" onClick={this.closeModal}>Close</button>
+        </Modal.Footer>
+      </Modal>
+    );
+  }
+  render() {
+    const { questions } = this.props;
+    const filteredQuestions = _.map(questions).filter(search =>
```
Comment (alan-maldonado) [inline src/components/ListQuestions.js]: It is an object where the key is the id. In the reducer I used _.mapKeys. It is easy to handle elements with object and key as the id of the element.

Comment (codinronan) [inline src/components/ListQuestions.js]: @alan-maldonado 👍  No problem. Let's go ahead and pull in that lodash package too then.

File: src/routes/routes_callbacks.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { fetchQuestions } from '../actions';
+
+export function onQuestionsEnter() {
+  const { questions: { data } } = store.getState();
+  if (_.isEmpty(data)) {
+    store.dispatch(fetchQuestions());
+  }
+}
+
+export const a = 'A';
```
Comment (alan-maldonado) [inline src/routes/routes_callbacks.js]: lol :D

**Comment by alan-maldonado** [comment]
@codinronan, I applied your changes. Thanks for the review

**Comment by codinronan** [comment]
@alan-maldonado merge it! :D

**Pull request merged**